pkg/sign/signer.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sign
+
+import (
+	"crypto"
+	"crypto/rand"
+	_ "crypto/sha256" // hash function will be used by Sign
+	_ "crypto/sha512" // hash function will be used by Sign
+	"errors"
+
+	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
+	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
+)
+
+type Signer interface {
+	Sign(data []byte) (*protobundle.Bundle, error)
+}
+
+// TODO: implement
+// type Fulcio struct {
+//	baseURL       string
+//	identityToken string
+// }
+
+// func (f *Fulcio) Sign(data []byte) (*protobundle.Bundle, error) {}
+
+type Keypair struct {
+	signer        crypto.Signer
+	hashAlgorithm protocommon.HashAlgorithm
+	publicKeyHint []byte
+}
+
+func SignerKeypair(s crypto.Signer, ha protocommon.HashAlgorithm, pkh []byte) Keypair {
```

Comment:
Do we want to make an options struct to initialize the signer? Then if we need to expand the options in the future, it won't require a breaking API change.